internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,18 @@
+package buffer
+
+type WriteBuffer interface {
+	// Create creates a buffer of 2*size passed as a parameter.
+	Create(size int) WriteBuffer
```

Comment:
I would suggest avoiding Create function from the interface as it'll vary a lot between the implementations. Interfaces should only define their core functionality, which in this case are Write, Upload, Status etc.

Reply:
I would suggest let's keep it like this for now. We can remove it from the interface later incase that doesn't fit while working on onDisk buffer.